ddtrace/internal/rate_limiter.py

```diff
@@ -18,6 +21,17 @@ class RateLimiter(_RateLimiter):
     def _has_been_configured(self):
         return self.rate_limit != DEFAULT_SAMPLING_RATE_LIMIT
 
+    def is_allowed(self, timestamp_ns: Optional[int] = None) -> bool:
+        if timestamp_ns is not None:
+            deprecate(
+                "The `timestamp_ns` parameter is deprecated and will be removed in a future version."
+                "Ratelimiter will use the current time.",
+                category=DDTraceDeprecationWarning,
+            )
+        # rate limits are tested and mocked in pytest so we need to compute the timestamp here
```

Comment:
an alternative, is just adding a method to the Python class for getting "now_ns", this could live in Rust, then we can still mock it in tests.

I don't feel too strongly, sort of same-same... but we can keep the Rust interface (RateLimiter) as requiring a timestamp being passed in, then we can have the Python interface (RateLimiterPy) generate the timestamp (if we do so via a method, then we should be able to mock it).

Reply:
This is not a blocking comment, resolve if you want.